src/metar/metar_csv2ioda.py

```diff
@@ -0,0 +1,322 @@
+
+                try:
+                    altim = float(row['Altimeter'])
+                    psfc = self.meteo_utils.altim_2_sfcPressure(altim, elev)
+                except:
+                    altim = netCDF4.default_fillvals['f4']
+                    psfc = netCDF4.default_fillvals['f4']
+
+                if ( (psfc is not netCDF4.default_fillvals['f4']) and (temp is not netCDF4.default_fillvals['f4']) and (dewp is not netCDF4.default_fillvals['f4']) ):
+                    spfh = self.meteo_utils.specific_humidity(dewp, psfc)
+                else:
+                    spfh = netCDF4.default_fillvals['f4']
+
+                data['ob_icao'].append(icao)
+                data['ob_time'].append(utime)
+                data['ob_datetime'].append(datetime.fromtimestamp(utime).strftime("%Y-%m-%dT%H:%M:%SZ"))
+                data['ob_lat'].append(lat)
+                data['ob_lon'].append(lon)
+                data['ob_elev'].append(elev)
+                data['ob_psfc'].append(psfc)
+                data['ob_temp'].append(temp)
+                data['ob_spfh'].append(spfh)
+                data['ob_uwnd'].append(uwnd)
+                data['ob_vwnd'].append(vwnd)
+
+        # Testing some utility functions to ensure they work properly.
+                '''
+                if ( (temp is not netCDF4.default_fillvals['f4']) and (dewp is not netCDF4.default_fillvals['f4']) ):
+                    tlcl = self.meteo_utils.t_lcl(temp, dewp)
+                    print ("DEBUG, Temp, Dewp, T_at_LCL is: " + str(temp) + ", " + str(dewp) + ", " + str(tlcl))
+                else:
+                    tlcl = netCDF4.default_fillvals['f4']
+
+                if ( (psfc is not netCDF4.default_fillvals['f4']) and (spfh is not netCDF4.default_fillvals['f4']) ):
+                    td = self.meteo_utils.t_dew(psfc, spfh)
+                    print ("DEBUG, re-derived dewpoint is: " + str(td))
+
+                if ( (psfc is not netCDF4.default_fillvals['f4']) and (temp is not netCDF4.default_fillvals['f4']) and (spfh is not netCDF4.default_fillvals['f4']) and (tlcl is not netCDF4.default_fillvals['f4']) ):
+                    the = self.meteo_utils.theta_e(psfc, temp, spfh, tlcl)
+                    print ("DEBUG, Theta-E is: " + str(the))
+
+                if ( (psfc is not netCDF4.default_fillvals['f4']) and (temp is not netCDF4.default_fillvals['f4']) and (tlcl is not netCDF4.default_fillvals['f4']) ):
+                    Th = temp * (100000.0/psfc)**(self.meteo_utils.R/self.meteo_utils.Cp)
+                    Plcl = 100000.0 * (tlcl/Th)**(self.meteo_utils.Cp/self.meteo_utils.R)
+                    the_lcl = self.meteo_utils.theta_e(Plcl, tlcl, spfh, tlcl)
+                    T500 = self.meteo_utils.compT_fr_The(the_lcl, 50000.0)
+                    print ("DEBUG, Parel T(500mb) is: " + str(T500) + ", " + str(Plcl) + ", " + str(Th))
+                '''
+
+        fh.close()
+
+        '''
```

Comment:
Remove this docstring of debug lines also

Reply:
A bit of conundrum in a way (to me at least).  I created all those added doc-block chunks to give 100% test coverage to the meteo_utils codes and show how they can be run using these data (METARs) as the example.  I realize it creates a large doc-block of unexecuted code, but thought it useful to others.

Would you suggest entire deletion or putting this chunk into a document or what?

Reply:
What about creating a separate unit test in the ioda converters just to test these (and other) utility functions?
Something that will check if, for example, a given T and P, make sure the output theta == the expected value, and so on. Can just print to a text file and check the output against a reference text file. This could be a new PR later.

Reply:
I went ahead and removed the doc-blocks, keeping them in a saved copy so I can refer back to them later.